client/dashboard/sites/settings-repositories/connect-repository-form.tsx

```diff
@@ -306,9 +332,25 @@ export const ConnectRepositoryForm = ( {
 	const isAdvancedSelected = formData.deploymentMode === 'advanced';
 
 	const handleChange = ( updates: Partial< ConnectRepositoryFormData > ) => {
+		let shouldRestoreInstallationFocus = false;
+		let shouldRestoreRepositoryFocus = false;
 		setFormData( ( prev ) => {
 			const newFormData = { ...prev, ...updates };
 
+			if (
+				'selectedInstallationId' in updates &&
+				updates.selectedInstallationId !== prev.selectedInstallationId
+			) {
+				shouldRestoreInstallationFocus = true;
+			}
+
+			if (
+				'selectedRepositoryId' in updates &&
+				updates.selectedRepositoryId !== prev.selectedRepositoryId
```

Comment:
When we keep the `updates.selectedRepositoryId !== prev.selectedRepositoryId` check in and the user selects the same repository again, we lose focus.


https://github.com/user-attachments/assets/b47aaa37-d41a-41cf-9953-9020a66d7c1b

Because of that, we could remove that check:

```suggestion
				'selectedRepositoryId' in updates
```

Reply:
This was a great catch, thanks for that! I have reproduced it locally, applying your suggestion 👍

Reply:
[2163953](https://github.com/Automattic/wp-calypso/pull/106521/commits/21639533f65cca12ad0c175056ae85b897a37fa3)